app/src/lib/stores/app-store.ts

```diff
@@ -1476,6 +1463,32 @@ export class AppStore extends TypedBaseStore<IAppState> {
     }
   }
 
+  private setupMergabilityPromise(
+    repository: Repository,
+    baseBranch: Branch,
+    compareBranch: Branch,
+    onLoad: (mergeTreeResult: MergeTreeResult | null) => void,
+    cleanup?: () => void
+  ) {
+    return promiseWithMinimumTimeout(
+      () => determineMergeability(repository, baseBranch, compareBranch),
+      500
+    )
+      .catch(err => {
+        log.warn(
+          `Error occurred while trying to merge ${baseBranch.name} (${baseBranch.tip.sha}) and ${compareBranch.name} (${compareBranch.tip.sha})`,
+          err
+        )
+        return null
+      })
+      .then(mergeStatus => {
+        onLoad(mergeStatus)
+      })
+      .finally(() => {
+        cleanup?.()
+      })
```

Comment:
This feels a little odd to me given that we're already returning a promise. The caller of this could tack on `.finally()` on the returned promise instead which would feel better to me given that the caller is the one introducing the side effect which needs to be reset.

Same goes for the `onLoad` callback, I think it would be more idiomatic to tack on a `.then` on the returned promise and have the `setupMergeabilityPromise` only concern itself with the reusable parts of promiseWithMinimumTimeout and error management.

Reply:
Yep. That totally makes sense. I originally pulled out a lot more logic that was in that method than whittled it down to return the promise.. as soon as I read this I was like.. should have whittled it down farther. :D